src/compiler/checker.ts

```diff
@@ -13175,6 +13177,21 @@ namespace ts {
             if (node.flags & NodeFlags.Abstract && node.body) {
                 error(node, Diagnostics.Method_0_cannot_have_an_implementation_because_it_is_marked_abstract, declarationNameToString(node.name));
             }
+
+            checkNameOfStaticMethodOrPropertyInClass(node);
+        }
+
+        function checkNameOfStaticMethodOrPropertyInClass(node: ClassElement) {

```

Comment:
I would consider this as grammar check -> like call it `checkGrammarStaticPropertyName`  and move it to the end of the line  [13161](https://github.com/Microsoft/TypeScript/pull/9215/files#diff-c3ed224e4daa84352f7f1abcd23e8ccaR13161)